File: C++/094_Binary Tree Inorder Traversal.cpp
class Solution {
public:
    vector<int> inorderTraversal(TreeNode *root) {
        vector<int> res;
        while (!stk.empty())
            stk.pop();
        while (1) {
            while (root) {
                stk.push(root);
                root = root->left;
            }
            if (stk.empty())
                break;
            root = stk.top();
            stk.pop();
            res.push_back(root->val);
            root = root->right;
        }
        return res;
    }
private:
    stack<TreeNode*> stk;
};

File: Python/094_Binary Tree Inorder Traversal.py
class Solution:
    # @param root, a tree node
    # @return a list of integers
    def inorderTraversal(self, root):
        res = []
        self._inOrder(root, res)
        return res

    def _inOrder(self, root, res):
        if not root:
            return
        self._inOrder(root.left, res)
        res.append(root.val)
        self._inOrder(root.right, res)